### ISSUE TITLE
md-list-icon aligment on lists

### ISSUE THREAD

**Issue by peluprvi** [angular/components#5090]
#### Bug, feature request, or proposal:
Proposal:
- Remove the `padding: 4px` of `.mat-list-icon`.
Requests:
- A list icon parent class with a fixed width.
- A separated icon background class for the list icon.

#### What is the expected behavior?
There are two specs for icon alignment: (1) using the raw icon and (2) using the icon inside a circle. In both, the space form the text to the left side of the list is the same (72dp). The only change is the icon width.
There is no padding on icon (1) ou (2) and the background is optional.
The first left pixel of the icon or the icon background is 16dp from the left of the list content.

Material specs:
https://material.io/guidelines/layout/structure.html#structure-side-nav
https://material.io/guidelines/components/lists.html#lists-specs

![](https://user-images.githubusercontent.com/1561360/27209732-e9198686-5223-11e7-9442-6721736461fe.png)

[Example 1: layout_structure_sidenav_structure1](https://user-images.githubusercontent.com/1561360/27047146-624d0072-4f7d-11e7-9c34-0ff6b7021e9b.png)
[Example 2: components_lists_keylines_single5](https://user-images.githubusercontent.com/1561360/27047154-695cf93a-4f7d-11e7-90ec-e4fdca5cd587.png)
[Example 3: components_lists_keylines_two10](https://user-images.githubusercontent.com/1561360/27047160-6cee187c-4f7d-11e7-8676-939d8b6a4a66.png)

#### What is the current behavior?
`md-list-icon` uses its own width as the space plus the icon and text paddings to create a 72dp width. This approach gives more than 16dp (currently it is 20dp) from icon to the left of the list content and the icon doesn't align with other vertical elements (like the user email on the specs above).

#### What are the steps to reproduce?
```html
<!-- How to align the texts? -->
<md-list>
  <md-list-item *ngFor="let message of messages">
    <md-icon md-list-icon>folder</md-icon>
    <h3 md-line> {{message.from}} </h3>
  </md-list-item>
</md-list>
<md-list>
  <md-list-item *ngFor="let message of messages">
    <h3 md-line> {{message.from}} </h3>
  </md-list-item>
</md-list>
```
```html
<!-- Solution example -->
<md-list>
  <md-list-item *ngFor="let message of messages">
    <!-- Fixed spacer -->
    <md-list-icon-spacer>
      <!-- Icon can have a md-list-icon-bg for rounded background -->
      <md-icon md-list-icon md-list-icon-bg>folder</md-icon>
    </md-list-icon-spacer>
    <h3 md-line> {{message.from}} </h3>
  </md-list-item>
</md-list>
<md-list>
  <md-list-item *ngFor="let message of messages">
    <!-- Fixed spacer -->
    <md-list-icon-spacer></md-list-icon-spacer>
    <h3 md-line> {{message.from}} </h3>
  </md-list-item>
</md-list>
```

#### What is the use-case or motivation for changing an existing behavior?
Create a sidenav with a user profile like the fist image above. Or align a list with icons with a list without icons.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
https://github.com/angular/material2/commit/b1a9cb549f46a8da988c0742e3f03b1dc07a7a85

#### Is there anything else we should know?
This needs to be considered for the right icon too. And on dense options.

**Comment by jelbourn** [comment]
Is `md-list-avatar` what you're looking for?

**Comment by PhilipSultanescu** [comment]
I agree the styling doesn't conform with the material design guidelines, the Expected values are correct.

In AngularJS Material you can find how it should look like:
https://material.angularjs.org/latest/demo/list

**Comment by Philip-Sultanescu** [comment]
What is the status on this? Will it be corrected?

**Comment by jelbourn** [comment]
The spec for lists was updated extensively in 2018 in such a way that I believe it makes this issue obsolete.

**Comment by philip-sultanescu** [comment]
The lists still don't follow the official specs, maybe there is another more up-to-date issue for this.

https://material.io/components/lists#specs
<img width="553" alt="Screenshot 2021-02-24 at 12 00 26" src="https://user-images.githubusercontent.com/74723618/108991172-08d1b600-7698-11eb-8e58-1ec9de029981.png">

https://material.io/components/navigation-drawer#specs
<img width="458" alt="Screenshot 2021-02-24 at 12 01 08" src="https://user-images.githubusercontent.com/74723618/108991176-0b341000-7698-11eb-89e4-89a125b24ec0.png">

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
